### PR TITLE
Buildkite pipeline unification fixes

### PR DESCRIPTION
**Pull request by oakrizan**
## Proposed commit message
In scope of https://github.com/elastic/ingest-dev/issues/3072:
- added notifications for pipeline steps
- removed top level notifications for pipelines
- added BK hooks script to changeset checks
- removed BEATS_PROJECT_NAME env var from auditbeat and heartbeat

BK Builds: https://buildkite.com/elastic/beats/builds/5305

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: .buildkite/x-pack/pipeline.xpack.winlogbeat.yml [hunk 1]
```diff
@@ -8,41 +8,156 @@ env:
...
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        env:
+          PLATFORMS: "+all linux/amd64 linux/arm64 windows/amd64 darwin/amd64 darwin/arm64"
+        notify:
+          - github_commit_status:
+              context: "Xpack/Winlogbeat Packaging"
+
```
Comment (dliappis) [inline .buildkite/x-pack/pipeline.xpack.winlogbeat.yml:163]: As mentioned in a previous comment, I think larger refactorings/fixes for x-pack/winlogbeat (or any other specific beat) belong in a separate PR.

This PR should only be fixing common *small* issues that apply to all migrated beats so far.

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -9,6 +9,7 @@ steps:
             - path:
                 - auditbeat/
                 - .buildkite/auditbeat/
+                - .buildkite/hooks/
```
Comment (dliappis) [inline .buildkite/pipeline.yml:12]: additionally we should update the diff command for `auditbeat` and `heartbeat` to look like

`git diff --name-only origin/${GITHUB_PR_TARGET_BRANCH}...HEAD`

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -94,13 +98,38 @@ steps:
...
+                  commit: "${BUILDKITE_COMMIT}"
+                  branch: "${BUILDKITE_BRANCH}"
+                  env:
+                    - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
+                    - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
+
   - label: "Trigger Deploy/K8S"
     plugins:
       - monorepo-diff#v1.0.1:
           diff: "git diff --name-only HEAD~1"
```
Comment (dliappis) [inline .buildkite/pipeline.yml:128]:
```suggestion
          diff: "git diff --name-only origin/${GITHUB_PR_TARGET_BRANCH}...HEAD"
```

**Review by dliappis — CHANGES_REQUESTED** [review]
Thanks for this clean up!

Left suggestions for a number of changes.

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (oakrizan) [reply .buildkite/pipeline.yml:13]: Done!

File: .buildkite/pipeline.yml [hunk 4]
```diff
@@ -94,13 +98,38 @@ steps:
...
+                - testing/**
+              config:
+                trigger: "beats-xpack-winlogbeat"
+                build:
+                  commit: "${BUILDKITE_COMMIT}"
+                  branch: "${BUILDKITE_BRANCH}"
+                  env:
+                    - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
+                    - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
+
```
[the remark this answers is not in the document]
Comment (oakrizan) [reply .buildkite/pipeline.yml:133]: It was there since the branch was "growing" from x-pack/winlogbeat which wasn't in main at that moment. Rebase solved the issue :)

File: .buildkite/x-pack/pipeline.xpack.winlogbeat.yml (hunk 1 above)
Comment (oakrizan) [reply .buildkite/x-pack/pipeline.xpack.winlogbeat.yml]: It was there since the branch was "growing" from x-pack/winlogbeat which wasn't in main at that moment. Rebase solved the issue :)

File: .buildkite/pipeline.yml (hunk 3 above)
Comment (oakrizan) [reply .buildkite/pipeline.yml]: Already in place.

File: .buildkite/heartbeat/heartbeat-pipeline.yml [hunk 5]
```diff
@@ -183,9 +174,9 @@ steps:
       - label: ":ubuntu: Packaging Linux X86"
         key: "heartbeat-package-linux-x86"
         env:
-          PLATFORMS: "${PACKAGING_PLATFORMS}"
+          PLATFORMS: "+all linux/amd64 linux/arm64 windows/amd64 darwin/amd64 darwin/arm64"
```
Comment (dliappis) [inline .buildkite/heartbeat/heartbeat-pipeline.yml:177]: Warning: when doing the backport for `7.17` we must remove `darwin/arm64`, see https://elastic.slack.com/archives/C0522G6FBNE/p1712924637322749

**Review by dliappis — APPROVED** [review]
LGTM (provided that the CI steps for the affected pipelines are green)

File: .buildkite/heartbeat/heartbeat-pipeline.yml (hunk 5 above)
Comment (oakrizan) [reply .buildkite/heartbeat/heartbeat-pipeline.yml:177]: Thanks, I'm aware of that
